src/test/rules/spaceInParensRuleTests.ts

```diff
@@ -0,0 +1,514 @@
+  { code: 'foo( (1 + 2) / 1)', options: ['never', { exceptions: ['()'] }] },
+  { code: 'foo( (\n1 + 2\n) )', options: ['never', { exceptions: ['()'] }] },
+
+  { code: 'foo()', options: ['always', { exceptions: ['empty'] }] },
+  { code: 'foo( )', options: ['always', { exceptions: ['{}'] }] },
+  { code: 'foo(\n1 + 2\n)', options: ['always', { exceptions: ['empty'] }] },
+  { code: 'foo()', options: ['never', { exceptions: ['{}'] }] },
+  { code: 'foo( )', options: ['never', { exceptions: ['empty'] }] },
+  { code: 'foo( \n1 + 2\n )', options: ['never', { exceptions: ['empty'] }] },
+
+  { code: 'foo({ bar: "baz" }, [ 1, 2 ])', options: ['always', { exceptions: ['{}', '[]'] }] },
+  { code: 'foo({\nbar: "baz"\n}, [\n1,\n2\n])', options: ['always', { exceptions: ['{}', '[]'] }] },
+  { code: 'foo(); bar({bar:"baz"}); baz([1,2])', options: ['always', { exceptions: ['{}', '[]', '()'] }] },
+  { code: 'foo( { bar: "baz" }, [ 1, 2 ] )', options: ['never', { exceptions: ['{}', '[]'] }] },
+  { code: 'foo( {\nbar: "baz"\n}, [\n1,\n2\n] )', options: ['never', { exceptions: ['{}', '[]'] }] },
+  { code: 'foo( ); bar( {bar:"baz"} ); baz( [1,2] )', options: ['never', { exceptions: ['{}', '[]', 'empty'] }] },
+
+  // faulty exceptions option
+  { code: 'foo( { bar: "baz" } )', options: ['always', { exceptions: [] }] }
+]);
+
+ruleTester.addTestGroup('invalid', 'should fail invalid', [
+  {
+    code: 'foo( bar)',
+    output: 'foo( bar )',
+    options: ['always'],
+    errors: expecting([{ message: MISSING_SPACE_ERROR, line: 0, column: 8 }])
+  },
+  {
+    code: 'foo(bar)',
+    output: 'foo( bar )',
+    options: ['always'],
+    errors: expecting([
+    { message: MISSING_SPACE_ERROR, line: 0, column: 4 },
+    { message: MISSING_SPACE_ERROR, line: 0, column: 7 }
+    ])
+  },
+  {
+    code: 'var x = ( 1 + 2) * 3',
+    output: 'var x = ( 1 + 2 ) * 3',
+    options: ['always'],
+    errors: expecting([{ message: MISSING_SPACE_ERROR, line: 0, column: 15 }])
+  },
+  {
+    code: 'var x = (1 + 2 ) * 3',
+    output: 'var x = ( 1 + 2 ) * 3',
+    options: ['always'],
+    errors: expecting([{ message: MISSING_SPACE_ERROR, line: 0, column: 9 }])
+  },
+  {
+    code: 'foo\n(bar\n)\n',
+    output: 'foo\n( bar\n)\n',
+    options: ['always'],
+    errors: expecting([{ message: MISSING_SPACE_ERROR, line: 1, column: 1 }])
+  },
+  {
+    code: 'bar(baz )',
+    output: 'bar(baz)',
+    options: ['never'],
+    errors: expecting([ { message: REJECTED_SPACE_ERROR, line: 0, column: 8 } ])
+  },
+  {
+    code: 'bar( baz )',
+    output: 'bar(baz)',
+    options: ['never'],
+    errors: expecting([
+    { message: REJECTED_SPACE_ERROR, line: 0, column: 4 },
+    { message: REJECTED_SPACE_ERROR, line: 0, column: 9 }
+    ])
+  },
+  {
+    code: 'var x = ( 4 + 5) * 6',
+    output: 'var x = (4 + 5) * 6',
+    options: ['never'],
+    errors: expecting([ { message: REJECTED_SPACE_ERROR, line: 0, column: 9 } ])
+  },
+  {
+    code: 'var x = (4 + 5 ) * 6',
```

Comment:
Lets duplicate one or two tests that have 2 or more extra spaces to make sure that the fixer works.